appoetry1/LoginViewController.swift

```diff
@@ -11,22 +11,63 @@ import UIKit
 class LoginViewController: UIViewController {
     @IBOutlet weak var emailLogin: UITextField!
     @IBOutlet weak var passwordLogin: UITextField!
-    
-    var showMeAgain: (() -> Void)?
-    
+
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        // Do any additional setup after loading the view.
     }
     
     @IBAction func pressLogin(_ sender: Any) {
-        NotificationCenter.default.post(name: Notification.Name("goToMain"), object: nil)
+
+        let providedEmailAddress = emailLogin.text
+        
+        let isEmailAddressValid = isValidEmailAddress(emailAddressString: providedEmailAddress!)
+        
+        if isEmailAddressValid {
+            
+            print("Email address is valid")
+        } else {
+            print("Email address is not valid")
+            displayAlertMessage(messageToDisplay: "Email address is not valid")
+        }
     }
     
+    func isValidEmailAddress(emailAddressString: String) -> Bool {
+        
+        var returnValue = true
```

Comment:
This kind of validation function that uses regex can be extracted even further to a static function somewhere in utilities outside of MVVM hierarchy. Only thing it does is check the validity of email anyway